velox/functions/prestosql/MapSubset.h

```diff
@@ -0,0 +1,221 @@
+        out_type<Map<TType, Generic<T1>>>& out,                     \
+        const arg_type<Map<TType, Generic<T1>>>& inputMap,          \
+        const arg_type<Array<TType>>& keys) {                       \
+      if (keys.empty()) {                                           \
+        return;                                                     \
+      }                                                             \
+                                                                    \
+      if (!constantSearchKeys_) {                                   \
+        searchKeys_.clear();                                        \
+        initializeSearchKeys(keys);                                 \
+      }                                                             \
+                                                                    \
+      if (searchKeys_.empty()) {                                    \
+        return;                                                     \
+      }                                                             \
+                                                                    \
+      auto toFind = searchKeys_.size();                             \
+                                                                    \
+      for (const auto& entry : inputMap) {                          \
+        if (!searchKeys_.contains(entry.first)) {                   \
+          continue;                                                 \
+        }                                                           \
+                                                                    \
+        if (!entry.second.has_value()) {                            \
+          auto& keyWriter = out.add_null();                         \
+          keyWriter = entry.first;                                  \
+        } else {                                                    \
+          auto [keyWriter, valueWriter] = out.add_item();           \
+          keyWriter = entry.first;                                  \
+          valueWriter.copy_from(entry.second.value());              \
+        }                                                           \
+                                                                    \
+        --toFind;                                                   \
+        if (toFind == 0) {                                          \
+          break;                                                    \
+        }                                                           \
+      }                                                             \
+    }                                                               \
+                                                                    \
+   private:                                                         \
+    void initializeSearchKeys(const arg_type<Array<TType>>& keys) { \
+      for (const auto& key : keys.skipNulls()) {                    \
```

Comment:
```
nit: searchKeys_.reserve(keys.size())
```

Reply:
In folly::F14FastSet, reserve is logically private (part of 'detail' namespace). I get a linter warning when trying to use it.